File: 1380.go
package leetcode

func luckyNumbers (matrix [][]int) []int {
	row := make([]int, len(matrix))
	col := make([]int, len(matrix[0]))
	for i := 0; i < len(matrix); i++ {
		min := matrix[i][0]
		for j := 1; j < len(matrix[0]); j++ {
			if (matrix[i][j] < min) {
				min = matrix[i][j]
			}
		}
		row[i] = min
	}
	for i := 0; i < len(matrix[0]); i++ {
		max := 0
		for j := 0; j < len(matrix); j++ {
			if (matrix[j][i] > max) {
				max = matrix[j][i]
			}
		}
		col[i] = max
	}
	var res []int
	for i := 0; i < len(matrix); i++ {
		for j := 0; j < len(matrix[0]); j++ {
			if (matrix[i][j] == row[i] && matrix[i][j] == col[j]) {
				res = append(res, matrix[i][j])
			}
		}
	}
	return res
}

File: 1380.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> luckyNumbers (vector<vector<int>>& matrix) {
        vector<int> row(matrix.size(), 0);
        for (int i = 0; i < matrix.size(); ++i) {
            int min = matrix[i][0];
            for (int j = 1; j < matrix[0].size(); ++j) {
                if (matrix[i][j] < min) {
                    min = matrix[i][j];
                }
            }
            row[i] = min;
        }
        vector<int> col(matrix[0].size(), 0);
        for (int i = 0; i < matrix[0].size(); ++i) {
            int max = 0;
            for (int j = 1; j < matrix.size(); ++j) {
                if (matrix[i][j] > max) {
                    max = matrix[i][j];
                }
            }
            col[i] = max;
        }
        vector<int> res;
        for (int i = 0; i < matrix.size(); ++i) {
            for (int j = 0; j < matrix[0].size(); ++j) {
                if (matrix[i][j] == row[i] && matrix[i][j] == col[j])
                    res.emplace_back(matrix[i][j]);
            }
        }
        return res;
    }
};